proto/api/api.proto

```diff
@@ -223,7 +226,12 @@ message SuggestGasPriceResponse {
   uint64 gasPrice = 1;
 }
 
+// To be deprecated
 message EstimateGasForActionRequest {
+  iotextypes.Action action = 1;
+}
+
+message EstimateGasForActionRequestNew {
   iotextypes.Execution execution = 1;
```

Comment:
@ququzone makes a good point that we need to make it general, so that it could be extended to more than execution. Can we use oneOf here?

Reply:
how about this https://github.com/iotexproject/iotex-proto/pull/29